File: src/main/java/com/akagiyui/drive/service/UploadService.java
package com.akagiyui.drive.service;

import com.akagiyui.drive.model.request.PreUploadRequest;
import org.springframework.web.multipart.MultipartFile;

/**
 * 上传 服务接口
 *
 * @author AkagiYui
 */
public interface UploadService {
    /**
     * 请求文件上传
     */
    void requestUpload(PreUploadRequest preUploadRequest);

    /**
     * 上传分片
     */
    void uploadChunk(String fileHash, MultipartFile chunk, String chunkHash, int chunkIndex);
}

File: app/src/main/kotlin/com/akagiyui/drive/service/UploadService.kt
package com.akagiyui.drive.service

import com.akagiyui.drive.entity.User
import com.akagiyui.drive.entity.UserFile
import com.akagiyui.drive.entity.cache.UploadTask
import com.akagiyui.drive.model.request.CreateUploadTaskRequest
import org.springframework.web.multipart.MultipartFile

/**
 * 上传 服务接口
 *
 * @author AkagiYui
 */
interface UploadService {
    /**
     * 创建上传任务
     */
    fun createUploadTask(user: User, createUploadTaskRequest: CreateUploadTaskRequest): UploadTask

    /**
     * 上传分片
     */
    fun uploadChunk(user: User, taskId: String, chunk: MultipartFile, chunkHash: String, chunkIndex: Int)

    /**
     * 接收文件
     */
    fun receiveMultipartFiles(user: User, files: List<MultipartFile>, folder: String?): List<UserFile>
}
